AlloyReact/Helpers/HtmlHelpers.cs

```diff
@@ -25,19 +25,20 @@ public static class HtmlHelpers
         /// Renders an HTML container in which a React component will be mounted
         /// </summary>
         /// <param name="html">Associated HTML helper</param>
+        /// <param name="propertyName">Name of the associated content property</param>
         /// <param name="component">Class or function name of React component</param>
         /// <param name="props">Optional object to pass as props to the component</param>
         /// <remarks>This HTML helper also ensure required script resources are loaded</remarks>
         /// <returns></returns>
-        public static IHtmlString ReactComponent(this HtmlHelper html, string component, object props = null)
+        public static IHtmlString ReactComponent(this HtmlHelper html, string propertyName, string component, object props = null)
```

Comment:
Maybe you can just use the existing `ReactComponentFor` instead? :)

Reply:
But then there should be no different between the output HTML in view or edit mode.

Reply:
Why do you want the `data-epi-*` attributes to show up on the public site?

Reply:
We need it in order to map a page propery to a component’s prop. We have a page propery named someNumber, and it can be anything that we don’t now at compile time. In the component, we have a prop named number. By returning the page property, we know what propery value should we get and map to the component’s prop.

Reply:
I'm sorry, I still don't understand how that is affected by not being in edit mode? I'd assume that a partner would not want the `data-epi-*` properties to be shown when viewing the site. To only want to use our OPE editing functionality when in OPE?

Reply:
Take a look at ReactNumber.jsx, this.props.propertyName is empty in View mode. And it was hard-coding the property name in props: props.someNumber.

'someNumber' is a property name, defined in the StandardPage model. I don't think we can use that fixed string in the component.

Reply:
Yes, but the editable HOC uses `props.propertyName` in order to retrieve data and set it back to the given `propName` on the component. In this case "number". You don't read the `data-epi-property-name` anywhere that I can see. So why would it be needed outside of edit mode?

Reply:
No I don't have to read the attribute, but props.propertyName will be null then. Without my changes, props.propertyName is available in Edit mode only.